server/game/cards/08.6-SAT/TheKingOnTheWall.js

```diff
@@ -0,0 +1,40 @@
+const DrawCard = require('../../drawcard.js');
+
+class TheKingOnTheWall extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction({ faction: 'targaryen', unique: true });
+        this.whileAttached({
+            effect: ability.effects.addTrait('King')
+        });
+
+        this.reaction({
+            when: {
+                afterChallenge: event =>
+                    event.challenge.winner === this.controller &&
+                    event.challenge.challengeType === 'power' &&
+                    event.challenge.isParticipating(this.parent) &&
+                    this.numberOfOutOfFactionLocations() > 0 &&
+                    this.controller.canDraw()
+            },
+            handler: () => {
+                var cards = this.numberOfOutOfFactionLocations();
+                cards = this.controller.drawCardsToHand(cards).length;
+
+                this.game.addMessage('{0} uses {1} to draw {2} {3}',
+                    this.controller, this, cards, cards > 1 ? 'cards' : 'card');
+            }
+        });
+    }
+
+    numberOfOutOfFactionLocations() {
```

Comment:
The contents of this method can be shortened to:

```javascript
return this.controller.getNumberOfCardsInPlay(card => card.getType() === 'location' && card.isOutOfFaction());
```

(It also makes sure neutral cards aren't counted as out of faction, as per the rules.)

Reply:
thanks! I looked for something like this before my (bad) implementation, but couldn't find it :)